lib/spack/spack/extensions.py

```diff
@@ -27,28 +65,26 @@ def extension_name(path):
         The extension name or None if path doesn't match the format
         for Spack's extension.
     """
-    regexp_match = re.search(extension_regexp, os.path.basename(path))
+    regexp_match = re.search(_extension_regexp, os.path.basename(path))
     if not regexp_match:
-        msg = "[FOLDER NAMING]"
-        msg += " {0} doesn't match the format for Spack's extensions"
-        tty.warn(msg.format(path))
-        return None
+        raise ExtensionNamingError(path)
     return regexp_match.group(1)
 
 
-def load_command_extension(command, path):
+def load_command_extension(command):
```

Comment:
Same as above - we are now raising and the docstring is not up to date.

Reply:
See above. I can fix the docstring.